File: app/src/main/java/com/kakeibo/ui/items/Balance.java
//package com.kakeibo.ui.items;
//
//import java.math.BigDecimal;
//import java.math.RoundingMode;
//
//public class Balance {
//    private BigDecimal income;
//    private BigDecimal expense;
//    private BigDecimal balance;
//
//    public static Balance newInstance(int fractionDigits) {
//        return new Balance(new BigDecimal(0).setScale(fractionDigits, RoundingMode.UNNECESSARY),
//                new BigDecimal(0).setScale(fractionDigits, RoundingMode.UNNECESSARY));
//    }
//
//    public Balance (BigDecimal income, BigDecimal expense) {
//        this.income = income;
//        this.expense = expense;
//    }
//
//    public BigDecimal getIncome () {
//        return income;
//    }
//
//    public BigDecimal getExpense () {
//        return expense;
//    }
//
//    public BigDecimal getBalance () {
//        balance = income.subtract(expense);
//        return balance;
//    }
//
//    public void addIncome (BigDecimal income) {
//        this.income = this.income.add(income);
//    }
//
//    public void addExpense (BigDecimal expense) {
//        this.expense = this.expense.add(expense);
//    }
//
//    public long inMinusOut () {
//        if (getBalance().compareTo(new BigDecimal(0)) < 0) return -1;
//        else if (getBalance().compareTo(new BigDecimal(0)) > 0) return 1;
//        else return 0;
//    }
//}

File: app/src/main/java/com/kakeibo/ui/model/Balance.kt
package com.kakeibo.ui.model

import java.math.BigDecimal
import java.math.RoundingMode

class Balance(var income: BigDecimal, var expense: BigDecimal) {
    var balance: BigDecimal? = null
        get() {
            field = income.subtract(expense)
            return field
        }
        private set

    fun addIncome(income: BigDecimal?) {
        this.income = this.income.add(income)
    }

    fun addExpense(expense: BigDecimal?) {
        this.expense = this.expense.add(expense)
    }

    fun inMinusOut(): Long {
        return if (balance!!.compareTo(BigDecimal(0)) < 0) -1 else if (balance!!.compareTo(BigDecimal(0)) > 0) 1 else 0
    }

    companion object {
        fun newInstance(fractionDigits: Int): Balance {
            return Balance(BigDecimal(0).setScale(fractionDigits, RoundingMode.UNNECESSARY),
                    BigDecimal(0).setScale(fractionDigits, RoundingMode.UNNECESSARY))
        }
    }
}